packages/box/src/Box.tsx

```diff
@@ -39,13 +39,13 @@ function InlineBox<TProps>(
 ): JSX.Element;
 
 function InlineBox(props: BoxProps, ref: React.Ref<any>) {
-  if (props.as !== undefined) {
+  if (props.as != null) {
     const { as: Component, ...rest } = props;
     // @ts-expect-error
     return <Component {...rest} ref={ref} />;
   }
 
-  if (props.href !== undefined) {
+  if (props.href != null) {
```

Comment:
can we go even so far as to make this`typeof props.href !== 'string'`?

Reply:
wfm

Reply:
actually, getting this from ts

```Types of property 'onCopy' are incompatible.
        Type '((event: ClipboardEvent<HTMLAnchorElement>) => void) | undefined' is not assignable to type '((event: ClipboardEvent<HTMLDivElement>) => void) | undefined'.
          Type '(event: ClipboardEvent<HTMLAnchorElement>) => void' is not assignable to type '(event: ClipboardEvent<HTMLDivElement>) => void'.
            Types of parameters 'event' and 'event' are incompatible.
              Type 'ClipboardEvent<HTMLDivElement>' is not assignable to type 'ClipboardEvent<HTMLAnchorElement>'.
                Type 'HTMLDivElement' is missing the following properties from type 'HTMLAnchorElement': charset, coords, download, hreflang, and 21 ```

Reply:
I can enforce `string` in other components (Menu, SubMenu and Button) but not in Box 🤔

Reply:
my guess is that
```js
if (typeof props.href != 'string') {
  return ...
}
// props is not being narrowed from `BoxAnchorDefault | BoxDivDefault` to `BoxDivDefault` anymore
```
but no biggie